### PR TITLE
feat(plugins): Support for remote extension definitions and resolving remote extension configuration type for remote plugins/extensions

### PR DESCRIPTION
**Pull request by jonsie**
Required for: https://github.com/spinnaker/orca/pull/3875

As I was implementing the remote extension/plugin in Orca I realized it would be very useful to be able to declare a bean that defines the remote extension so we can resolve the remote extension configuration type (and anything else that may need to be resolved in the future).

### CODE REVIEW

File: kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/extension/RemoteExtension.kt [hunk 1]
```diff
@@ -46,11 +45,16 @@ class RemoteExtension(
...
    * prior to the remote extension invocation.
    */
-  val config: Map<String, Any>,
+  val config: RemoteExtensionConfigType,
 
   private val transport: RemoteExtensionTransport
 ) {
 
+  @Suppress("UNCHECKED_CAST")
+  fun <T: RemoteExtensionConfigType> getTypedConfig(): T {
```
Comment (jonsie) [inline kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/extension/RemoteExtension.kt:54]: This works, but I feel like there may be a better way.

File: kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/RemotePlugin.kt [hunk 2]
```diff
@@ -18,6 +18,7 @@
 package com.netflix.spinnaker.kork.plugins.remote
 
 import com.netflix.spinnaker.kork.annotations.Beta
+import com.netflix.spinnaker.kork.plugins.remote.extension.RemoteExtension
```
Comment (robzienert) [inline kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/RemotePlugin.kt:21]: nit: Looks like we can clean this import up.

File: kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/extension/RemoteExtensionDefinition.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.netflix.spinnaker.kork.plugins.remote.extension
+
+/**
+ * Define properties of the remote extension. Spinnaker services must implement this and provide it
+ * as a Spring bean in order for remote plugin extensions to be resolved from the configuration.
+ */
+interface RemoteExtensionDefinition {
```
Comment (robzienert) [inline kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/extension/RemoteExtensionDefinition.kt:7]: If the service is implementing this, would it not be a `RemoteExtensionPointDefinition`?

File: kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/RemotePlugin.kt (hunk 2 above)
Comment (jonsie) [reply kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/RemotePlugin.kt:21]: There's a small package refactor in this PR - this is the result of that.

File: kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/extension/RemoteExtensionDefinition.kt (hunk 3 above)
Comment (jonsie) [reply kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/remote/extension/RemoteExtensionDefinition.kt:7]: Dig it.

**Pull request merged**